firmware/Core/Src/littlefs/flash_driver.c

```diff
@@ -4,7 +4,15 @@
 #include "littlefs/flash_driver.h"
 #include "debug_tools/debug_uart.h"
 
+#include "config/static_config.h"
+#ifndef FLASH_ENABLE_UART_DEBUG_PRINT
+    #error "FLASH_ENABLE_UART_DEBUG_PRINT not defined"
```

Comment:
Just read the comment explaining this above. The #error confused me, though, because a common idiom is that simply #define'ing toggles the compilation. It seems error prone and cumbersome to have to remember to include this guard. 

Suggest to make this clear in the coding style doc.

Reply:
Hmm that's fair. The fear was that if you forget to include the header file in the place you're using the configuration variable, then it takes on a value silently. The idea was that, by requiring a check that the macro is present (regardless of its value), it prevents that issue. 

It's an annoying quirk of C that `#if` assumes a default value instead of erroring when the macro is unavailable.

Thinking about it more, `static const` variables are probably the best approach in achieving the safety I was going for. I'll switch to that.